server/config.go

```diff
@@ -0,0 +1,185 @@
+package server
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/flashbots/go-boost-utils/types"
+	"io/ioutil"
+)
+
+type rawConfiguration struct {
+	FeeRecipient          string `json:"fee_recipient"`
+	ValidatorRegistration struct {
+		BuilderRelays []string `json:"builder_relays"`
+		Enabled       bool     `json:"enabled"`
+		GasLimit      string   `json:"gas_limit"`
+	} `json:"validator_registration"`
+}
+
+type rawConfigurationFile struct {
+	BuilderRelaysGroups map[string][]string `json:"builder_relays_groups"`
+
+	ProposerConfig map[string]rawConfiguration `json:"proposer_config"`
+	DefaultConfig  rawConfiguration            `json:"default_config"`
+}
+
+// newRawConfigurationFile creates a temporary rawConfigurationFile used to build a
+// ProposerConfigurationStorage, by reading content from a JSON file.
+func newRawConfigurationFile(filename string) (*rawConfigurationFile, error) {
+	// Read JSON file content.
+	bytes, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+
+	// Tries to unmarshal content in JSON.
+	placeholder := &rawConfigurationFile{}
+	if err := json.Unmarshal(bytes, placeholder); err != nil {
+		return nil, err
+	}
+
+	return placeholder, nil
+}
+
+// ConfigurationStorage holds one proposer configuration.
+type ConfigurationStorage struct {
```

Comment:
maybe `ProposerRelayConfig` could be a better name

Reply:
Renamed in ce4e6dcd160154120f6bdabd57751cc445913b6d.